backend/src/subscriptions/dependencies.py

```diff
@@ -0,0 +1,25 @@
+from http import HTTPStatus
+from fastapi import Depends, HTTPException
+from sqlalchemy import select
+from sqlalchemy.orm import selectinload
+from src.auth.dependencies import get_current_user
+from src.common.dependencies import get_session
+from src.subscriptions.models import Subscription
+
+
+def retrieve_subscription(
+    id: int,
+    user=Depends(get_current_user),
+    session=Depends(get_session),
+) -> Subscription:
+    subscription = session.scalar(
+        select(Subscription)
+        .where(Subscription.id == id)
+        .options(
+            selectinload(Subscription.user.and_(Subscription.user_id == user.id)),
```

Comment:
```suggestion
            selectinload(Subscription.user),
```

not necessary, when you saw that line its usually for cases like event has 10 likes (user actually not part of the relationship), only one of them belong the user and i don't want to show the other 9

in this case user is part of the relationship, and that would be part of the join condition already

Reply:
Thanks! have made the suggested change